gtda/mapper/cover.py

```diff
@@ -374,9 +378,11 @@ class CubicalCover(BaseEstimator, TransformerMixin):
 
     """
 
-    _hyperparameters = {'kind': [str, ['uniform', 'balanced']],
-                        'n_intervals': [int, (1, np.inf)],
-                        'overlap_frac': [float, (0, 1)]}
+    _hyperparameters = {
+        'kind': {'type': str, 'in': ['uniform', 'balanced']},
+        'n_intervals': {'type': int, 'in': Interval(1, np.inf, closed='left')},
+        'overlap_frac': {'type': float, 'in': Interval(0, 1, closed='neither')}
```

Comment:
Since the parameters are passed to `OneDCover`, this is a  bit different than `metric_params`  and i wonder if it makes sense to check  those at all. Or, call the check from `OneDCover`?

Reply:
I think we don't actually repeat the checks here because we don't call the `fit` method of `OneDimensionalCover` but rather go straight to hidden `_fit`-type methods which do not perform validation. I'm not sure I understand the parallel with `metric_params`, though.

Reply:
So, 'conceptually', what i am calling out, is that if each class had a method `validate_params`, `CubicalCover.validate_params()`  could potentially call `OneDimensionalCover.validate_params()`.
The analogy with metrics, is that there, you never instantiate a featurisation transformer, and only call the method meaning that the same trick does not work.
My bad though, I overlooked that we call the hidden methods directly, meaning that we need to `validate_params` beforehand even for `OneDimensionalCover`